builtin/aws/lambda/releaser.go

```diff
@@ -0,0 +1,323 @@
+			default:
+				step.Update("Error creating permissions: %q, %q", aerr.Code(), aerr.Message())
+				return err
+			}
+		} else {
+			return err
+		}
+	} else {
+		step.Update("Created permissions for public access to the lambda URL")
+	}
+	step.Done()
+
+	step = sg.Add("Creating Lambda URL...")
+	defer step.Abort()
+
+	cfo, err := lambdasrv.CreateFunctionUrlConfig(&createFunctionUrlConfigInput)
+	if err != nil {
+		log.Error("Error creating function url config", "error", err)
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "ResourceConflictException":
+				// function url config already exists. get it. maybe safe to continue
+				step.Update("Function url config already exists")
+
+				// retrieve existing function url to update state
+				if gfc, err := lambdasrv.GetFunctionUrlConfig(&lambda.GetFunctionUrlConfigInput{
+					FunctionName: aws.String(dep.FuncArn),
+				}); err != nil {
+					// this should not realistically occur
+					log.Error("Error getting function url config", "error", err)
+					return err
+				} else {
+					state.Url = *gfc.FunctionUrl
+					step.Update("Reusing existing Lambda URL: %q", state.Url)
+				}
+			default:
+				step.Update("Error creating function url config: %q, %q", aerr.Code(), aerr.Message())
+				return err
+			}
+		} else {
+			return err
+		}
+	} else {
+		// update state
+		state.Url = *cfo.FunctionUrl
+		step.Update("Created Lambda URL: %q", state.Url)
+	}
+
+	step.Done()
+
+	return nil
+}
+
+func (r *Releaser) Release(
+	ctx context.Context,
+	log hclog.Logger,
+	src *component.Source,
+	ui terminal.UI,
+	dep *Deployment,
+) (*Release, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	// Create our resource manager and create
+	rm := r.resourceManager(log)
+	if err := rm.CreateAll(
+		ctx, log, sg, ui, src,
+		dep,
+	); err != nil {
+		log.Info("Error creating resources", "error", err)
+		return nil, err
+	}
+
+	// Get our function url state to verify
+	fnUrlState := rm.Resource("function_url").State().(*Resource_FunctionUrl)
+	log.Info("Function URL state", "url", fnUrlState.Url)
+	if fnUrlState == nil {
+		return nil, status.Errorf(codes.Internal, "function url state is nil, this should never happen")
+	}
+
+	return &Release{
+		Url:           fnUrlState.Url,
+		FuncArn:       dep.FuncArn,
+		VerArn:        dep.VerArn,
+		ResourceState: rm.State(),
+	}, nil
+}
+
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	rm := r.resourceManager(log)
+
+	// Destroy All
+	return rm.DestroyAll(ctx, log, sg, ui, sess)
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// "AWS_IAM" or "NONE"
+	AuthType string `hcl:"auth_type,optional"`
+	// todo(kevinwang): CORS
```

Comment:
Yea, I was hoping to add `CORS` config as an _improvement_ in a follow up PR. Is that reasonable?

Reply:
Very reasonable! Sounds good. I'd say if you plan on getting to it immediately, we can leave this be and wait for the follow-up PR. If it might be a few weeks or so, I'd open a GitHub issue about it so we don't lose track of the work.